cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -170,3 +140,67 @@ func fixConfiguration(config cascfg.Configuration) cascfg.Configuration {
 	config.ProtoVersion = 4
 	return config
 }
+
+type esSpanHandlerBuilder struct {
+	logger        *zap.Logger
+	client        es.Client
+	configuration escfg.Configuration
+}
+
+func newESBuilder(config *escfg.Configuration, logger *zap.Logger) *esSpanHandlerBuilder {
+	return &esSpanHandlerBuilder{
+		configuration: *config,
+		logger:        logger,
+	}
+}
+
+func (e *esSpanHandlerBuilder) BuildHandlers() (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error) {
+	client, err := e.getClient()
+	if err != nil {
+		return nil, nil, err
+	}
+	spanStore := esSpanstore.NewSpanWriter(client, e.logger)
+
+	return buildHandlers(spanStore, e.logger, nil)
+}
+
+func (e *esSpanHandlerBuilder) getClient() (es.Client, error) {
+	if e.client == nil {
+		client, err := e.configuration.NewClient()
+		e.client = client
+		return e.client, err
+	}
+	return e.client, nil
+}
+
+func buildHandlers(
+	spanStore spanstore.Writer,
+	logger *zap.Logger,
+	metricsFactory metrics.Factory,
+) (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error) {
+	// TODO: currently ES has no metricsFactory, hence the workaround. Once ES adds a metricsFactory, this nullpointer check should be unnecessary.
```

Comment:
nit: we have a NullFactory which can be passed around instead. However given we'll be landing the metrics PR pretty soon, this is fine.